engines/buried/graphics.cpp

```diff
@@ -0,0 +1,884 @@
+	    REQUIRED(0x00A0),     REQUIRED(0x00A1),     REQUIRED(0x00A2),     REQUIRED(0x00A3),
+	    REQUIRED(0x00A4),     REQUIRED(0x00A5),     REQUIRED(0x00A6),     REQUIRED(0x00A7),
+	    REQUIRED(0x00A8),     REQUIRED(0x00A9),     REQUIRED(0x00AA),     REQUIRED(0x00AB),
+	    REQUIRED(0x00AC),     REQUIRED(0x00AD),     REQUIRED(0x00AE),     REQUIRED(0x00AF),
+	    REQUIRED(0x00B0),     REQUIRED(0x00B1),     REQUIRED(0x00B2),     REQUIRED(0x00B3),
+	    REQUIRED(0x00B4),     REQUIRED(0x00B5),     REQUIRED(0x00B6),     REQUIRED(0x00B7),
+	    REQUIRED(0x00B8),     REQUIRED(0x00B9),     REQUIRED(0x00BA),     REQUIRED(0x00BB),
+	    REQUIRED(0x00BC),     REQUIRED(0x00BD),     REQUIRED(0x00BE),     REQUIRED(0x00BF),
+	    REQUIRED(0x00C0),     REQUIRED(0x00C1),     REQUIRED(0x00C2),     REQUIRED(0x00C3),
+	    REQUIRED(0x00C4),     REQUIRED(0x00C5),     REQUIRED(0x00C6),     REQUIRED(0x00C7),
+	    REQUIRED(0x00C8),     REQUIRED(0x00C9),     REQUIRED(0x00CA),     REQUIRED(0x00CB),
+	    REQUIRED(0x00CC),     REQUIRED(0x00CD),     REQUIRED(0x00CE),     REQUIRED(0x00CF),
+	    REQUIRED(0x00D0),     REQUIRED(0x00D1),     REQUIRED(0x00D2),     REQUIRED(0x00D3),
+	    REQUIRED(0x00D4),     REQUIRED(0x00D5),     REQUIRED(0x00D6),     REQUIRED(0x00D7),
+	    REQUIRED(0x00D8),     REQUIRED(0x00D9),     REQUIRED(0x00DA),     REQUIRED(0x00DB),
+	    REQUIRED(0x00DC),     REQUIRED(0x00DD),     REQUIRED(0x00DE),     REQUIRED(0x00DF),
+	    REQUIRED(0x00E0),     REQUIRED(0x00E1),     REQUIRED(0x00E2),     REQUIRED(0x00E3),
+	    REQUIRED(0x00E4),     REQUIRED(0x00E5),     REQUIRED(0x00E6),     REQUIRED(0x00E7),
+	    REQUIRED(0x00E8),     REQUIRED(0x00E9),     REQUIRED(0x00EA),     REQUIRED(0x00EB),
+	    REQUIRED(0x00EC),     REQUIRED(0x00ED),     REQUIRED(0x00EE),     REQUIRED(0x00EF),
+	    REQUIRED(0x00F0),     REQUIRED(0x00F1),     REQUIRED(0x00F2),     REQUIRED(0x00F3),
+	    REQUIRED(0x00F4),     REQUIRED(0x00F5),     REQUIRED(0x00F6),     REQUIRED(0x00F7),
+	    REQUIRED(0x00F8),     REQUIRED(0x00F9),     REQUIRED(0x00FA),     REQUIRED(0x00FB),
+	    REQUIRED(0x00FC),     REQUIRED(0x00FD),     REQUIRED(0x00FE),     REQUIRED(0x00FF)
+};
+
+#undef REQUIRED
+#undef NOT_REQUIRED
+
+#endif // USE_ICONV
+
+Graphics::Font *GraphicsManager::createFont(int size, bool bold) const {
+	// MS Gothic for the Japanese version
+	// Arial or Arial Bold for everything else
+	if (_vm->getLanguage() == Common::JA_JPN)
+		return createMSGothicFont(size, bold);
+
+	return createArialFont(size, bold);
+}
+
+Graphics::Font *GraphicsManager::createArialFont(int size, bool bold) const {
+	Common::SeekableReadStream *stream = findArialStream(bold);
+
+	if (!stream)
+		error("Failed to find Arial%s font", bold ? " Bold" : "");
+
+	// Map the heights needed to point sizes
+	if (bold) {
+		if (size != 20)
+			error("Unhandled Arial Bold height %d", size);
+
+		size = 12;
+	} else {
+		switch (size) {
+		case 12:
+		case 13:
+			size = 7;
+			break;
+		case 14:
+			size = 8;
```

Comment:
The part of the following commit that makes use of the new font size mode introduced in #639 and fixes usage of an inccorect DPI value should probably be merged to fix font rendering issues.
https://github.com/project-cabal/cabal/commit/457a23b7ec8f04e7b1a39d95ed8077c64942cda1